engine/resource/src/test/test_resource_archive.cpp

```diff
@@ -858,6 +858,37 @@ TEST(dmResourceArchive, LoadFromDisk_Compressed)
     dmResourceArchive::Delete(archive);
 }
 
+
+static dmResourceArchive::Result TestDecryption(void* buffer, uint32_t buffer_len)
```

Comment:
Looking at it now, I wonder if the path of the resource is required?
(E.g. for doing different encrypting based on the file type)
E.g. skipping encryption for larger files like textures and sounds.

Reply:
That is a good point and perhaps something to add as part of a new task or based on user requests!